crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -3160,3 +3327,19 @@ fn find_conflicting_extra(conflicting: &Conflicts, reqs: &[Requirement]) -> Opti
     }
     None
 }
+
+#[derive(Debug, Default, Clone)]
+struct ConflictTracker {
+    /// How often a decision on the package was discarded due to another package decided earlier.
+    affected: FxHashMap<Id<PubGrubPackage>, usize>,
+    /// Package(s) to be prioritized after the next unit propagation
+    ///
+    /// Distilled from `affected` for fast checking in the hot loop.
+    priotize: Vec<Id<PubGrubPackage>>,
+    /// How often a package was decided earlier and caused another package to be discarded.
+    culprit: FxHashMap<Id<PubGrubPackage>, usize>,
+    /// Package(s) to be de-prioritized after the next unit propagation
+    ///
+    /// Distilled from `culprit` for fast checking in the hot loop.
+    depriotize: Vec<Id<PubGrubPackage>>,
```

Comment:
One thing to note here is that `Id<PubGrubPackage>` is different for proxies, like `Package::Extra` and `Package::Package` have different IDs.

Reply:
Surprisingly, keeping them separate showed better performance than conflict counts over the proxies of a package.